File: AbbreviateWordName/Abbreviate_two_word_name.py

def abbrev_name(name):
    '''
    a function to convert a name into initials. This kata strictly takes two words with one 
    space in between them.The output should be two capital letters with a dot separating them.
    It should look like this:

    Sam Harris => S.H
    patrick feeney => P.F
    '''
    split_name = name.split(" ")
    return f'{split_name[0][0]}.{split_name[1][0]}'.upper()

# alternative soln:


def abbrevName(name):
    return '.'.join(w[0] for w in name.split()).upper()

# tests
# import codewars_test as test

# try:
#     from solution import abbrevName as abbrev_name
# except ImportError:
#     from solution import abbrev_name

# @test.describe("Fixed Tests")
# def basic_tests():
#     @test.it('Basic Test Cases')
#     def basic_test_cases():
#         test.assert_equals(abbrev_name("Sam Harris"), "S.H")
#         test.assert_equals(abbrev_name("patrick feenan"), "P.F")
#         test.assert_equals(abbrev_name("Evan C"), "E.C")
#         test.assert_equals(abbrev_name("P Favuzzi"), "P.F")
#         test.assert_equals(abbrev_name("David Mendieta"), "D.M")

File: AbbreviateWordName/AbbreviateTwoWordName.js
function abbrevName(name) {
  let splitName = name.split(' ');
  return `${splitName[0][0]}.${splitName[1][0]}`.toUpperCase();
}

// alternative solns
const abbrevName = (name) => name.match(/\b\w/g).join('.').toUpperCase();

function abbrevName(name) {
  return name
    .split(' ')
    .map((x) => x.substr(0, 1).toUpperCase())
    .join('.');
}

// tests
/*
const chai = require("chai");
const assert = chai.assert;
chai.config.truncateThreshold=0;

describe("Basic tests", () => {
  it("Testing for fixed tests", () => {
    assert.strictEqual(abbrevName("Sam Harris"), "S.H");
    assert.strictEqual(abbrevName("Patrick Feenan"), "P.F");
    assert.strictEqual(abbrevName("Evan Cole"), "E.C");
    assert.strictEqual(abbrevName("P Favuzzi"), "P.F");
    assert.strictEqual(abbrevName("David Mendieta"), "D.M");
  });
});

*/
